asg0_1.c

```diff
@@ -0,0 +1,124 @@
+#include <stdio.h>
+#include <limits.h>
+int isNumber(char currChar){
+    if(currChar<='9'&&currChar>='0'){
+        return 1;
+    }
+    return 0;
+
+}
+
+int isSpace(char currChar){
+    if(currChar==' '){
+        return 1;
+    }
+    return 0;
+
+}
+
+int checkOperator(char currChar){
+    if(currChar == '+' || currChar== '-' || currChar == '*' || currChar == '/'){
+        return 1;
+    }
+    return 0;
+
+}
+
+int getNumber(char* currChar, int* index) {
+    int tempNumber = 0;
+    while (isNumber(currChar[*index])==1) {
+        tempNumber = tempNumber * 10 + (currChar[*index] - '0');
+        (*index)++;
+    }
+    return tempNumber;
+
+}
+
+int evaluateResult(int operands[],char operators[], int operandInd){
+    int ans = operands[0];
+    int index = 1; 
+    for (int i = 0; i < operandInd; i++) {
+        if (operators[i] == '+') {
+            ans += operands[index];
+        } else {
+            ans -= operands[index];
+        }
+        index++;
+    }
+    return ans;
+
+}
+
+int calculate(char* inputExp) {
+    int operands[10000] = {0}, invalidInputError = INT_MAX, divByZero = INT_MIN, numIndex = 0, operandInd = 0, index = 0, inputIndex=0;
+    char operators[10000];
+
+    while(inputExp[inputIndex]){
+        if(isSpace(inputExp[inputIndex])==1 || isNumber(inputExp[inputIndex])==1 || checkOperator(inputExp[inputIndex])) inputIndex++;
+        else return invalidInputError;
+    }
+
+    while (inputExp[index]) {
+        if (isSpace(inputExp[index])==1) index++;
+        else if (isNumber(inputExp[index])==1) {
+            operands[numIndex++] = getNumber(inputExp, &index);

```

Comment:
this kind of formatting is not readable, learn about formatting, or use vscode formatting